packages/@lwc/ssr-runtime/src/index.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+const MULTI_SPACE = /\s+/g;
+
+type Attributes = Record<string, string>;
+
+type LightningElementConstructor = typeof LightningElement;
+
+class ClassList {
+    el: LightningElement;
+
+    constructor(el: LightningElement) {
+        this.el = el;
+    }
+
+    add(...newClassNames: string[]) {
+        const className = this.el.className;
+        const set = new Set(className.split(MULTI_SPACE).filter(Boolean));
+        for (const newClassName of newClassNames) {
+            set.add(newClassName);
+        }
+        this.el.className = Array.from(set).join(' ');
+    }
+
+    contains(className: string) {
+        const currentClassNameStr = this.el.className;
+        return currentClassNameStr.split(MULTI_SPACE).includes(className);
+    }
+
+    remove(...classNamesToRemove: string[]) {
+        const className = this.el.className;
+        const set = new Set(className.split(MULTI_SPACE).filter(Boolean));
+        for (const newClassName of classNamesToRemove) {
+            set.delete(newClassName);
+        }
+        this.el.className = Array.from(set).join(' ');
+    }
+
+    replace(oldClassName: string, newClassName: string) {
+        let classWasReplaced = false;
+        const className = this.el.className;
+        const listOfClasses = className.split(MULTI_SPACE).filter(Boolean);
+        listOfClasses.forEach((value, idx) => {
+            if (value === oldClassName) {
+                classWasReplaced = true;
+                listOfClasses[idx] = newClassName;
+            }
+        });
+        this.el.className = listOfClasses.join(' ');
+        return classWasReplaced;
+    }
+
+    toggle(classNameToToggle: string, force?: boolean) {
+        const classNameStr = this.el.className;
+        const set = new Set(classNameStr.split(MULTI_SPACE).filter(Boolean));
+        if (!set.has(classNameToToggle) && force !== false) {
+            set.add(classNameToToggle);
+        } else if (set.has(classNameToToggle) && force !== true) {
+            set.delete(classNameToToggle);
+        }
+        this.el.className = Array.from(set).join(' ');
+        return set.has(classNameToToggle);
+    }
+}
+
+export class LightningElement {
+    static renderMode?: 'light' | 'shadow';
+
+    isConnected = false;
+    className = '';
+    // TODO [W-14977927]: protect internals from userland
+    __attrs?: Record<string, string | true>;
```

Comment:
or use this for `Attributes` above. There's three different types for attributes 🫣